Installation issue: python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf

### Steps to reproduce the issue

```console
$ spack install hpl %aocc +openmp ^amdblis threads=openmp ^openmpi fabrics=cma,ucx
or
$ spack install tk tcl %aocc

both fail in step install python-3-13.2-5
...
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Installing python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf [44/59]
==> No binary for python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf found: installing from source
==> Using cached archive: /home/mmps/heckes/spack/var/spack/cache/_source-cache/archive/b8/b8d79530e3b7c96a5cb2d40d431ddb512af4a563e863728d8713039aa50203f9.tgz
==> Ran patch() for python
==> python: Executing phase: 'configure'
==> python: Executing phase: 'build'
==> python: Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    '/home/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/gcc-11.4.1/gmake-4.4.1-gmkxr6fg43pagt2sgsu3s6xgxynrxmro/bin/make' '-j1' 'install'

4 errors found in build log:
     1320    if test $? -ne 0 ; then \
     1321       echo "generate-posix-vars failed" ; \
     1322       rm -f ./pybuilddir.txt ; \
     1323       exit 1 ; \
     1324    fi
     1325    LD_LIBRARY_PATH=/tmp/mmps/spack-stage/spack-stage-python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/spack-sr
             c ./python -E -c 'import sys ; from sysconfig import get_platform ; print("%s-%d.%d" % (get_platform(), *
             sys.version_info[:2]))' >platform
  >> 1326    [ERROR] _tkinter failed to import: /tmp/mmps/spack-stage/spack-stage-python-3.13.2-5iwdtwaejt7qtwkanrx4ty
             6omdf7urkf/spack-src/build/lib.linux-x86_64-3.13/_tkinter.cpython-313-x86_64-linux-gnu.so: undefined symb
             ol: Tcl_FindExecutable
     1327    The necessary bits to build these optional modules were not found:
     1328    _dbm
     1329    To find the necessary bits, look in configure.ac and config.log.
     1330    
     1331    Following modules built successfully but were removed because they could not be imported:
     1332    _tkinter

     ...

     1373       if test -d "libpython3.13.so.dSYM"; then \
     1374               echo  /libpython3.13.so.1.0; \
     1375           /libpython3.13.so.1.0; \
     1376       fi \
     1377    fi
     1378    /home/mmps/heckes/spack/lib/spack/env/aocc/clang -shared -L/home/mmps/heckes/spack/opt/spack/linux-almali
             nux9-zen4/gcc-11.4.1/bzip2-1.0.8-24v3ya2k57mbbj43eqdceq33fzh5r4z5/lib -L/home/mmps/heckes/spack/opt/spack
             /linux-almalinux9-zen4/aocc-4.2.0/expat-2.6.4-i33afbypblaqflwakt2dcwk7hcp2jehb/lib -L/home/mmps/heckes/sp
             ack/opt/spack/linux-almalinux9-zen4/gcc-11.4.1/gdbm-1.23-x6r7esejhjlecfzel2s6xabx3n6oaq6m/lib -L/usr/lib6
             4  -L/home/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/aocc-4.2.0/libffi-3.4.6-dychgxhnrfrmsyvqat45
             kyddubm2uqru/lib -L/home/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/gcc-11.4.1/ncurses-6.5-2ddptx3
             fh66mc4e44nvib4hxkieo7y45/lib -L/home/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/gcc-11.4.1/openss
             l-3.3.1-nfzqtvlqkht4jmkafxvcb4el26vvljoy/lib64 -L/home/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/
             gcc-11.4.1/readline-8.2-f75clvqjaggp2jxfoswlocqm4zyqy4so/lib -L/home/mmps/heckes/spack/opt/spack/linux-al
             malinux9-zen4/aocc-4.2.0/sqlite-3.46.0-2flqlydetcli73q23feapfgrxix4toee/lib -L/home/mmps/heckes/spack/opt
             /spack/linux-almalinux9-zen4/aocc-4.2.0/util-linux-uuid-2.40.2-vjkskmwvwd4464tisf2pq3y3wfdatm5p/lib -L/ho
             me/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/gcc-11.4.1/xz-5.4.6-74br2l44nnff5mv25opaltdvpwehlzpk
             /lib -L/home/mmps/heckes/spack/opt/spack/linux-almalinux9-zen4/gcc-11.4.1/zlib-ng-2.2.1-eaxxa7zp2yezfiog7
             wy7b2d6vhasui4j/lib     Modules/_tkinter.o Modules/tkappinit.o   -o Modules/_tkinter.cpython-313-x86_64-l
             inux-gnu.so
  >> 1379    [ERROR] _tkinter failed to import: /tmp/mmps/spack-stage/spack-stage-python-3.13.2-5iwdtwaejt7qtwkanrx4ty
             6omdf7urkf/spack-src/build/lib.linux-x86_64-3.13/_tkinter.cpython-313-x86_64-linux-gnu.so: undefined symb
             ol: Tcl_FindExecutable
     1380    The necessary bits to build these optional modules were not found:
     1381    _dbm
     1382    To find the necessary bits, look in configure.ac and config.log.
     1383    
     1384    Following modules built successfully but were removed because they could not be imported:
     1385    _tkinter

     ...

     6137    /usr/bin/install -c -m 755 Modules/_curses_panel.cpython-313-x86_64-linux-gnu.so /home/mmps/heckes/spack/
             opt/spack/linux-almalinux9-zen4/aocc-4.2.0/python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/lib/python3.13/
             lib-dynload/_curses_panel.cpython-313-x86_64-linux-gnu.so
     6138    /usr/bin/install -c -m 755 Modules/_sqlite3.cpython-313-x86_64-linux-gnu.so /home/mmps/heckes/spack/opt/s
             pack/linux-almalinux9-zen4/aocc-4.2.0/python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/lib/python3.13/lib-d
             ynload/_sqlite3.cpython-313-x86_64-linux-gnu.so
     6139    /usr/bin/install -c -m 755 Modules/_ssl.cpython-313-x86_64-linux-gnu.so /home/mmps/heckes/spack/opt/spack
             /linux-almalinux9-zen4/aocc-4.2.0/python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/lib/python3.13/lib-dynlo
             ad/_ssl.cpython-313-x86_64-linux-gnu.so
     6140    /usr/bin/install -c -m 755 Modules/_hashlib.cpython-313-x86_64-linux-gnu.so /home/mmps/heckes/spack/opt/s
             pack/linux-almalinux9-zen4/aocc-4.2.0/python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/lib/python3.13/lib-d
             ynload/_hashlib.cpython-313-x86_64-linux-gnu.so
     6141    /usr/bin/install -c -m 755 Modules/_uuid.cpython-313-x86_64-linux-gnu.so /home/mmps/heckes/spack/opt/spac
             k/linux-almalinux9-zen4/aocc-4.2.0/python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/lib/python3.13/lib-dynl
             oad/_uuid.cpython-313-x86_64-linux-gnu.so
     6142    /usr/bin/install -c -m 755 Modules/_tkinter.cpython-313-x86_64-linux-gnu.so /home/mmps/heckes/spack/opt/s
             pack/linux-almalinux9-zen4/aocc-4.2.0/python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/lib/python3.13/lib-d
             ynload/_tkinter.cpython-313-x86_64-linux-gnu.so
  >> 6143    /usr/bin/install: cannot stat 'Modules/_tkinter.cpython-313-x86_64-linux-gnu.so': No such file or directo
             ry
  >> 6144    make: *** [Makefile:2309: sharedinstall] Error 1

See build log for details:
  /tmp/mmps/spack-stage/spack-stage-python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf/spack-build-out.txt
</pre></details>


### Information on your system

$ spack debug report 
* **Spack:** 1.0.0.dev0 (9213bf591907696e6150dc880b5c415c35b13513)
* **Python:** 3.9.21
* **Platform:** linux-almalinux9-zen5

$ spack maintainers python
scheibelp  skosukhin


cat /etc/os-release
NAME="AlmaLinux"
VERSION="9.5 (Teal Serval)"
ID="almalinux"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.5"
PLATFORM_ID="platform:el9"
PRETTY_NAME="AlmaLinux 9.5 (Teal Serval)"
ANSI_COLOR="0;34"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:almalinux:almalinux:9::baseos"
HOME_URL="https://almalinux.org/"
DOCUMENTATION_URL="https://wiki.almalinux.org/"
BUG_REPORT_URL="https://bugs.almalinux.org/"

ALMALINUX_MANTISBT_PROJECT="AlmaLinux-9"
ALMALINUX_MANTISBT_PROJECT_VERSION="9.5"
REDHAT_SUPPORT_PRODUCT="AlmaLinux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.5"
SUPPORT_END=2032-06-01

git log
commit 9213bf591907696e6150dc880b5c415c35b13513 (HEAD -> develop, origin/develop, origin/HEAD)
Author: Vicente Bolea <vicente.bolea@kitware.com>
Date:   Fri Feb 14 17:14:04 2025 -0500

    vtk: fix 9.4.1 concretization (#48946)
    
    * seacas: conflict 2024-06-27 with windows
    
    * vtk: fix 9.4.1 seacas dependency


HW-info:
Architecture:             x86_64
  CPU op-mode(s):         32-bit, 64-bit
  Address sizes:          52 bits physical, 57 bits virtual
  Byte Order:             Little Endian
CPU(s):                   512
  On-line CPU(s) list:    0-511
Vendor ID:                AuthenticAMD
  Model name:             AMD EPYC 9745 128-Core Processor
    CPU family:           26
    Model:                17
    Thread(s) per core:   2
    Core(s) per socket:   128
    Socket(s):            2
    Stepping:             0
    Frequency boost:      enabled
    CPU(s) scaling MHz:   65%
    CPU max MHz:          3707.8120
    CPU min MHz:          1500.0000
    BogoMIPS:             4799.67
    Flags:                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse 
                          sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good amd_lbr_v2 nopl no
                          nstop_tsc cpuid extd_apicid aperfmperf rapl pni pclmulqdq monitor ssse3 fma cx16 pcid sse4_1 
                          sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand lahf_lm cmp_legacy svm extapic cr8_legac
                          y abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt tce topoext perfctr_core perfctr_nb
                           bpext perfctr_llc mwaitx cpb cat_l3 cdp_l3 hw_pstate ssbd mba perfmon_v2 ibrs ibpb stibp ibr
                          s_enhanced vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx
                          512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt x
                          savec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local avx_vnni avx512_bf16 c
                          lzero irperf xsaveerptr rdpru wbnoinvd amd_ppin cppc arat npt lbrv svm_lock nrip_save tsc_sca
                          le vmcb_clean flushbyasid decodeassists pausefilter pfthreshold avic v_vmsave_vmload vgif x2a
                          vic v_spec_ctrl vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni 
                          avx512_bitalg avx512_vpopcntdq la57 rdpid bus_lock_detect movdiri movdir64b overflow_recov su
                          ccor smca fsrm avx512_vp2intersect flush_l1d debug_swap amd_lbr_pmc_freeze
Virtualization features:  
  Virtualization:         AMD-V
Caches (sum of all):      
  L1d:                    12 MiB (256 instances)
  L1i:                    8 MiB (256 instances)
  L2:                     256 MiB (256 instances)
  L3:                     512 MiB (16 instances)
NUMA:                     
  NUMA node(s):           8
  NUMA node0 CPU(s):      0-31,256-287
  NUMA node1 CPU(s):      32-63,288-319
  NUMA node2 CPU(s):      64-95,320-351
  NUMA node3 CPU(s):      96-127,352-383
  NUMA node4 CPU(s):      128-159,384-415
  NUMA node5 CPU(s):      160-191,416-447
  NUMA node6 CPU(s):      192-223,448-479
  NUMA node7 CPU(s):      224-255,480-511
Vulnerabilities:          
  Gather data sampling:   Not affected
  Itlb multihit:          Not affected
  L1tf:                   Not affected
  Mds:                    Not affected
  Meltdown:               Not affected
  Mmio stale data:        Not affected
  Reg file data sampling: Not affected
  Retbleed:               Not affected
  Spec rstack overflow:   Not affected
  Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
  Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
  Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; STIBP always-on; RSB filling; PBRSB-
                          eIBRS Not affected; BHI Not affected
  Srbds:                  Not affected
  Tsx async abort:        Not affected


### Additional information

[spack-stage-python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf-spack-build-out.txt](https://github.com/user-attachments/files/18815049/spack-stage-python-3.13.2-5iwdtwaejt7qtwkanrx4ty6omdf7urkf-spack-build-out.txt)

I don't set-up an environment

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/python/package.py
+++ b/repos/spack_repo/builtin/packages/python/package.py
@@ -639,11 +639,11 @@ class Python(Package):
                     ),
                 ]
             )
-            
+
         # Disable tkinter module in the configure script for Python 3.12 onwards if ~tkinter
         if spec.satisfies("@3.12:") and spec.satisfies("~tkinter"):
             config_args.append("py_cv_module__tkinter=n/a")
-            
+
         # Disable the nis module in the configure script for Python 3.11 and 3.12. It is deleted
         # in Python 3.13. See ``def patch`` for disabling the nis module in Python 3.10 and older.
         if spec.satisfies("@3.11:3.12"):

--- a/repos/spack_repo/builtin/packages/python/package.py
+++ b/repos/spack_repo/builtin/packages/python/package.py
@@ -639,6 +639,7 @@ class Python(Package):
                     ),
                 ]
             )
+            
         # Disable tkinter module in the configure script for Python 3.12 onwards if ~tkinter
         if spec.satisfies("@3.12:") and spec.satisfies("~tkinter"):
             config_args.append("py_cv_module__tkinter=n/a")

--- a/repos/spack_repo/builtin/packages/python/package.py
+++ b/repos/spack_repo/builtin/packages/python/package.py
@@ -639,7 +639,10 @@ class Python(Package):
                     ),
                 ]
             )
-
+        # Disable tkinter module in the configure script for Python 3.12 onwards if ~tkinter
+        if spec.satisfies("@3.12:") and spec.satisfies("~tkinter"):
+            config_args.append("py_cv_module__tkinter=n/a")
+            
         # Disable the nis module in the configure script for Python 3.11 and 3.12. It is deleted
         # in Python 3.13. See ``def patch`` for disabling the nis module in Python 3.10 and older.
         if spec.satisfies("@3.11:3.12"):

